src/app/patient-creation/patient-creation.component.spec.ts

```diff
@@ -20,34 +34,144 @@ import { PatientResourceService } from '../openmrs-api/patient-resource.service'
 import { PatientRelationshipTypeService } from '../patient-dashboard/common/patient-relationships/patient-relation-type.service';
-import { PatientRelationshipTypeResourceService } from '../openmrs-api/patient-relationship-type-resource.service';
 import { PatientEducationService } from '../etl-api/patient-education.service';
 
+const testLocations = [
+  {
+    name: 'Test A',
+    stateProvince: 'Foo',
+    uuid: 'uuid1'
+  },
+  {
+    name: 'Test B',
+    stateProvince: 'Bar',
+    uuid: 'uuid2'
+  }
+];
+
+const testConcept = {
+  answers: [
+    { display: 'TEACHER', uuid: 'teacher-uuid' },
+    { display: 'CASUAL WORKER', uuid: 'casual-worker-uuid' },
+    { display: 'HEALTH WORKER', uuid: 'health-worker-uuid' }
+  ],
+  name: {
+    display: 'OCCUPATION',
+    uuid: 'test-occupation-concept-uuid'
+  },
+  uuid: 'test-uuid'
+};
+
+const testEducationLevels = [
+  {
+    uuid: 'a899e0ac-1350-11df-a1f1-0026b9348838',
+    display: 'NONE'
+  },
+  {
+    uuid: 'a8afe910-1350-11df-a1f1-0026b9348838',
+    display: 'PRIMARY SCHOOL'
+  },
+  {
+    uuid: 'a8afe9d8-1350-11df-a1f1-0026b9348838',
+    display: 'SECONDARY SCHOOL'
+  },
+  {
+    uuid: 'a89e4728-1350-11df-a1f1-0026b9348838',
+    display: 'UNIVERSITY'
+  }
+];
+
+const testIdentifierTypes = [
+  { uuid: 'uuid1', name: 'A test ID' },
+  { uuid: 'uuid2', name: 'A sophisticated test ID' },
+  { uuid: 'uuid3', name: 'An everyday test ID' }
+];
+
+const testRelationshipTypes = [
+  { uuid: 'uuid1', aIsToB: 'Parent', bIsToA: 'Child', display: 'Parent/Child' }
+];
+
+const conceptResourceServiceStub = {
+  getConceptByUuid: (uuid) => of(testConcept)
+};
+
+const locationResourceServiceStub = {
+  getLocations: () => of(testLocations)
+};
+
+const patientEducationServiceStub = {
+  getEducationLevels: () => testEducationLevels
+};
+
+const patientIdentifierTypeServiceStub = {
+  getPatientIdentifierTypes: () => of(testIdentifierTypes)
+};
+
+const patientRelationshipTypeServiceStub = {
+  getRelationshipTypes: () => of(testRelationshipTypes)
+};
+
+const patientCreationServiceStub = {
+  getpatientResults: () => of([]),
+  generateIdentifier: (userId) => {
+    return of({ identifier: '123456789-0' });
+  },
+  patientResults: () => of([]),
+  searchPatient: (searchString) => of([])
+};
+
+const userServiceStub = {
+  getLoggedInUser: () => {
+    const openmrsModel = {
+      systemId: 'test-12345'
+    };
+    return {
+      display: 'Test User',
+      openmrsModel: openmrsModel
+    };
+  }
+};
+
 describe('Component: Patient Creation Unit Tests', () => {
-  let fakeAppFeatureAnalytics: AppFeatureAnalytics, component;
+  let component: PatientCreationComponent;
+  let fixture: ComponentFixture<PatientCreationComponent>;
+  let debugElement: DebugElement;
+  let nativeElement: HTMLElement;
+  let patientCreationResourceService: PatientCreationResourceService;
+  let savePatientSpy: jasmine.Spy;
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
```

Comment:
Can we have a common component/file/class that configures the Testing module out of the box?

Reply:
That's definitely worth considering given the enormous amount of boilerplate we have to write for tests. Though I'd imagine that goes beyond this scope.